nosnoc/problem.py

```diff
@@ -456,12 +467,26 @@ def create_complementarity(self, x: List[ca.SX], y: ca.SX, sigma: ca.SX, tau: ca
 
         return
 
-    def create_complementarity_constraints(self, sigma_p: ca.SX, tau: ca.SX) -> None:
+    def create_complementarity_constraints(self, sigma_p: ca.SX, tau: ca.SX, Uk: ca.SX) -> None:
         opts = self.opts
+        # handle path complementarities TODO maintain sparsity?
```

Comment:
I guess for sparsity, the n_s loop could go down, but we are using a sparse solver anyway for now. So maybe just remove the todo.